### PR TITLE
NestedListWeightSum2 with HashMap

### PR DESCRIPTION
**Pull request by ErdemT09**
In this solution, in addition to keeping track of the maximum depth of the nested integers, we also sum individual layers and store these in a map. At the end, instead of re-looping through the nested integers, we just add the layer-sums elements in the map with their distance from the maximum depth.

### CODE REVIEW

File: src/main/java/algorithms/curated170/medium/NestedListWeightSum2HashMap.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    public int depthSumInverse(List<NestedInteger> nestedList) {
+        maxDepth(nestedList, 1);
+        return sumMapTotal();
+    }
+
+    private void maxDepth(List<NestedInteger> nInt, int depth) {
+        for (NestedInteger nestedInteger : nInt) {
+            if (nestedInteger.isInteger()) {
+                deepSumMap.put(depth, deepSumMap.get(depth)+nestedInteger.getInteger());
```
Comment (altayhunoglu) [inline src/main/java/algorithms/curated170/medium/NestedListWeightSum2HashMap.java:18]: Can we alter this for the null-check:
                deepSumMap.put(depth, deepSumMap.getOrDefault(depth, 0)+nestedInteger.getInteger());

Comment (ErdemT09) [reply src/main/java/algorithms/curated170/medium/NestedListWeightSum2HashMap.java:18]: Yes, null-checks or other checks like these shouldn't be forgotten.

Comment (ErdemT09) [reply src/main/java/algorithms/curated170/medium/NestedListWeightSum2HashMap.java]: Added.

**Review by altayhunoglu — APPROVED** [review]
The code is quite clear and smart. Thanks.

**Pull request merged**